diff --git a/classes/subscription/form/SubscriptionForm.inc.php b/classes/subscription/form/SubscriptionForm.inc.php
index 1c352c3166..11f9d0f13c 100644
--- a/classes/subscription/form/SubscriptionForm.inc.php
+++ b/classes/subscription/form/SubscriptionForm.inc.php
@@ -55,6 +55,7 @@ class SubscriptionForm extends Form {
 		foreach ($isoCodes->getCountries() as $country) {
 			$this->validCountries[$country->getAlpha2()] = $country->getLocalName();
 		}
+		HookRegistry::call('IsoCodesFactory::getCountries', [&$validCountries]);
 		asort($this->validCountries);
 
 		// User is provided and valid
diff --git a/plugins/reports/subscriptions/SubscriptionReportPlugin.inc.php b/plugins/reports/subscriptions/SubscriptionReportPlugin.inc.php
index 40ba5d1401..e3c0eace2a 100644
--- a/plugins/reports/subscriptions/SubscriptionReportPlugin.inc.php
+++ b/plugins/reports/subscriptions/SubscriptionReportPlugin.inc.php
@@ -146,6 +146,7 @@ class SubscriptionReportPlugin extends ReportPlugin {
 							$country = $countries->getByAlpha2($user->getCountry());
 						}
 						$columns[$index] = $country?$country->getLocalName():'';
+						HookRegistry::call('IsoCodesFactory::getLocalName', [&$columns[$index], $user->getCountry()]);
 						break;
 					case 'email':
 						$columns[$index] = $user->getEmail();
@@ -241,6 +242,7 @@ class SubscriptionReportPlugin extends ReportPlugin {
 					case 'country':
 						$country = $countries->getByAlpha2($user->getCountry());
 						$columns[$index] = $country?$country->getLocalName():'';
+						HookRegistry::call('IsoCodesFactory::getLocalName', [&$columns[$index], $user->getCountry()]);
 						break;
 					case 'email':
 						$columns[$index] = $user->getEmail();
